### PR TITLE
Fix77: integrate MKL/LAPACK subroutines

### PR DESCRIPTION
**Pull request by Altaimorin**
This branch is to integrate MKL and LAPACK subroutines into QUICK as an option for diagonalizations. Users need to load MKL or OPENBLAS modules so that $MKLROOT or $EBROOTOPENBLAS can be detected, otherwise the conventional subroutine for diagonalizations will be used. No library paths need be specified for linking the librarys.

Example for compilation on ICER HPCC:
module purge
module load intel/2020a    (imkl is loaded as a part of the toolchain)
make distclean
./configure --mkl intel

### CODE REVIEW

File: src/subs/diagmkl.f90 [hunk 1]
```diff
@@ -0,0 +1,43 @@
+SUBROUTINE DIAGMKL(NDIM,A,EVAL1,EVEC1,IERROR)
```
Comment (Madu86) [inline src/subs/diagmkl.f90:1]: Add copyrights with your information and describe what the subroutine does. See src/cuda/mgpu.h for an example.

File: configure [hunk 2]
```diff
@@ -984,12 +984,22 @@ for buildtype in $buildtypes; do
   # For serial and mpi versions, diagonalizers from lapack/mkl may be used
   if [ "$buildtype" = 'serial' -o "$buildtype" = 'mpi' ]; then
     if [ "$lapack" = 'yes' ]; then
-      fort_flags="$fort_flags -DLAPACK"
-      ldflags="$fort_flags -lopenblas"
-    elif [ "$mkl" = 'yes' ]; then
-      fort_flags="$fort_flags -DMKL"
-      ldflags="$fort_flags -L$INTELHOME -lmkl_intel_lp64 -lmkl_intel_thread -lmkl_core -liomp5 -lpthread"
-    fi
+		if [[ ! -n $EBROOTOPENBLAS ]]; then
```
Comment (Madu86) [inline configure:987]: Please fix this as follows. 

`if [ -z "$EBROOTOPENBLAS" ]; then`

File: configure [hunk 3]
```diff
@@ -984,12 +984,22 @@ for buildtype in $buildtypes; do
...
+			echo "OPENBLAS is not loaded, use conventional diagnolization"
```
Comment (Madu86) [inline configure:988]: echo "OPENBLAS is not loaded. Inbuilt diagonalizer will be used."

File: configure [hunk 4]
```diff
@@ -984,12 +984,22 @@ for buildtype in $buildtypes; do
...
+		else
+			echo "OPENBLAS is loaded. Use LAPACK acceleration of diagnolization"
```
Comment (Madu86) [inline configure:990]: echo "OPENBLAS is loaded. Diagonalizer from LAPACK will be used."

File: configure [hunk 5]
```diff
@@ -984,12 +984,22 @@ for buildtype in $buildtypes; do
...
+      		fort_flags="$fort_flags -DLAPACK"
+	  		ldflags="$ldflags -lopenblas"
+    	fi
+	elif [ "$mkl" = 'yes' ]; then
+		if [[ ! -n $MKLROOT ]]; then
+			echo "MKL is not loaded, use conventional diagnolization"
```
Comment (Madu86) [inline configure:996]: echo "MKL is not loaded. Inbuilt diagonalizer will be used."

File: configure [hunk 6]
```diff
@@ -984,12 +984,22 @@ for buildtype in $buildtypes; do
...
-    fi
+		if [[ ! -n $EBROOTOPENBLAS ]]; then
+			echo "OPENBLAS is not loaded, use conventional diagnolization"
+		else
+			echo "OPENBLAS is loaded. Use LAPACK acceleration of diagnolization"
+      		fort_flags="$fort_flags -DLAPACK"
+	  		ldflags="$ldflags -lopenblas"
+    	fi
+	elif [ "$mkl" = 'yes' ]; then
+		if [[ ! -n $MKLROOT ]]; then
```
Comment (Madu86) [inline configure:995]: `if [ -z "$MKLROOT" ]; then`

File: configure [hunk 7]
```diff
@@ -984,12 +984,22 @@ for buildtype in $buildtypes; do
...
+			echo "MKL is not loaded, use conventional diagnolization"
+	  	else
+	  		echo "MKL is loaded. Use MKL acceleration of diagnolization"
```
Comment (Madu86) [inline configure:998]: echo "MKL is loaded. Diagonalizer from MKL will be used."

**Review by Madu86 — CHANGES_REQUESTED** [review]
@Altaimorin I have some minor suggestions. Please check them.

File: src/subs/diagmkl.f90 (hunk 1 above)
Comment (Altaimorin) [reply src/subs/diagmkl.f90:1]: Added

File: configure (hunk 2 above)
Comment (Altaimorin) [reply configure:987]: Fixed

File: configure (hunk 3 above)
Comment (Altaimorin) [reply configure:988]: Fixed

File: configure (hunk 4 above)
Comment (Altaimorin) [reply configure:990]: Fixed

File: configure (hunk 5 above)
Comment (Altaimorin) [reply configure:996]: Fixed

File: configure (hunk 6 above)
Comment (Altaimorin) [reply configure:995]: Fixed

File: configure (hunk 7 above)
Comment (Altaimorin) [reply configure:998]: Fixed

**Pull request merged**